### PR TITLE
[kf5jobwidgets] new port

### PR DESCRIPTION
**Pull request by wrobelda**
**Describe the pull request**

- #### What does your PR fix?  
  Adds new port

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  All official.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes.

### CODE REVIEW

File: ports/kf5jobwidgets/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO KDE/kjobwidgets
+    REF v5.84.0
+    SHA512 57aa8d5756e1c9ce955975bfb5dc33ed78aaa984420effbb3b79984f9fd46794f3da092185bbcefe0464228d8f504349f7efd3a0ae79406954dc9ea84b3553f8
+    HEAD_REF master
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
```
Comment (JonLiu1993) [inline ports/kf5jobwidgets/portfile.cmake:11]:
```suggestion
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"

```

File: ports/kf5jobwidgets/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    OPTIONS
+        -DBUILD_TESTING=OFF
+    MAYBE_UNUSED_VARIABLES
+        -DBUILD_HTML_DOCS=OFF
+        -DBUILD_MAN_DOCS=OFF
+        -DBUILD_QTHELP_DOCS=OFF
+)
+
+vcpkg_install_cmake()
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/KF5JobWidgets)
```
Comment (JonLiu1993) [inline ports/kf5jobwidgets/portfile.cmake:21]:
```suggestion
vcpkg_cmake_install()
vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/KF5JobWidgets)

```

File: ports/kf5jobwidgets/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
+endif()
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/etc)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
+
+file(INSTALL ${SOURCE_PATH}/LICENSES/ DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright)
```
Comment (JonLiu1993) [inline ports/kf5jobwidgets/portfile.cmake:32]:
```suggestion
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/etc")
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")

file(INSTALL "${SOURCE_PATH}/LICENSES" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```

File: ports/kf5jobwidgets/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "kf5jobwidgets",
+  "version-string": "5.84.0",
```
Comment (JonLiu1993) [inline ports/kf5jobwidgets/vcpkg.json:3]:
```suggestion
  "version-semver": "5.84.0",
```

File: ports/kf5jobwidgets/vcpkg.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  "dependencies": [
+    "ecm",
+    "kf5coreaddons",
+    "kf5widgetsaddons",
+    "qt5-base",
+    {
+      "name": "qt5-x11extras",
+      "platform": "linux"
+    }
+  ]
```
Comment (JonLiu1993) [inline ports/kf5jobwidgets/vcpkg.json:15]:
```suggestion
      "platform": "linux"
    },
    {
      "name": vcpkg-cmake,
      "host": true
    },
    {
      "name": vcpkg-cmake-config,
      "host": true
    }
  ]
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/kf5jobwidgets/portfile.cmake (hunk 3 above)
Comment (wrobelda) [reply ports/kf5jobwidgets/portfile.cmake]: ${SOURCE_PATH}/LICENSES/ is a folder, not a file, and it contains several SPDX license files which all need to be copied to "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright" folder. The syntax I used does exactly that.

**Comment by wrobelda** [comment]
@JonLiu1993 all fixed now

File: ports/kf5jobwidgets/portfile.cmake (hunk 3 above)
Comment (JonLiu1993) [reply ports/kf5jobwidgets/portfile.cmake]: Sorry,I made a mistake

File: ports/kf5jobwidgets/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    DISABLE_PARALLEL_CONFIGURE
+    OPTIONS
+        -DBUILD_TESTING=OFF
+        -DBUILD_HTML_DOCS=OFF
+        -DBUILD_MAN_DOCS=OFF
+        -DBUILD_QTHELP_DOCS=OFF
+    MAYBE_UNUSED_VARIABLES
+        BUILD_HTML_DOCS
+        BUILD_MAN_DOCS
+        BUILD_QTHELP_DOCS
```
Comment (JonLiu1993) [inline ports/kf5jobwidgets/portfile.cmake:20]: @wrobelda ,I checked CMakeLists.txt, these options do not exist,

Comment (wrobelda) [reply ports/kf5jobwidgets/portfile.cmake:20]: @JonLiu1993, as mentioned in another PR, these are standard KDE variables used across KDE/QT software. If at any point a library adds a manual or HTML docs, these will apply. I thought that it makes sense for consistency with other frameworks where such variables are already present in their CMakeLists.txt.

Comment (wrobelda) [reply ports/kf5jobwidgets/portfile.cmake]: I removed them based on the discussion here: #19245

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**